src/macaron/slsa_analyzer/package_registry/maven_central_registry.py

```diff
@@ -152,41 +182,45 @@ def is_detected(self, build_tool: BaseBuildTool) -> bool:
         compatible_build_tool_classes = [Maven, Gradle]
         return any(isinstance(build_tool, build_tool_class) for build_tool_class in compatible_build_tool_classes)
 
-    def find_publish_timestamp(self, group_id: str, artifact_id: str, version: str | None = None) -> datetime:
+    def find_publish_timestamp(self, purl: str, registry_url: str | None = None) -> datetime:
         """Make a search request to Maven Central to find the publishing timestamp of an artifact.
 
-        If version is not provided, the timestamp of the latest version will be returned.
+        The reason for directly fetching timestamps from Maven Central is that deps.dev occasionally
+        misses timestamps for Maven artifacts, making it unreliable for this purpose.
 
         To see the search API syntax see: https://central.sonatype.org/search/rest-api-guide/
 
         Parameters
         ----------
-        group_id : str
-            The group id of the artifact.
-        artifact_id: str
-            The artifact id of the artifact.
-        version: str | None
-            The version of the artifact.
+        purl: str
+            The Package URL (purl) of the package whose publication timestamp is to be retrieved.
+            This should conform to the PURL specification.
+        registry_url: str | None
+            The registry URL that can be set for testing.
 
         Returns
         -------
         datetime
-            The artifact publish timestamp as a timezone-aware datetime object.
+            A timezone-aware datetime object representing the publication timestamp
+            of the specified package.
 
         Raises
         ------
         InvalidHTTPResponseError
-            If the HTTP response is invalid or unexpected.
+            If the URL construction fails, the HTTP response is invalid, or if the response
+            cannot be parsed correctly, or if the expected timestamp is missing or invalid.
         """
-        query_params = [f"q=g:{group_id}", f"a:{artifact_id}"]
-        if version:
-            query_params.append(f"v:{version}")
+        try:
+            purl_object = PackageURL.from_string(purl)
+        except ValueError as error:
+            logger.debug("Could not parse PURL: %s", error)
+        query_params = [f"q=g:{purl_object.namespace}", f"a:{purl_object.name}", f"v:{purl_object.version}"]
```

Comment:
`purl_object.version` has type `str | None`. If it's none the last query param would be `'v:None'`. I don't think it has the same behavior as the old implementation. In the old implementation, if version is None, no extra query param is added. I wonder if this new behavior is intended?

Reply:
See commit [17f3453](https://github.com/oracle/macaron/pull/872/commits/17f3453c4be75784f00a422b71ea829e3f04e4de).